ocrd_utils/ocrd_utils/str.py

```diff
@@ -85,6 +85,9 @@ def make_file_id(ocrd_file, output_file_grp):
             except ValueError:
                 n = len(ids)
             ret = concat_padded(output_file_grp, n)
+            while next(ocrd_file.mets.find_files(ID=ret), None):
+                n += 1
+                ret = concat_padded(output_file_grp, n)
```

Comment:
I thought we agreed we don't want to evade clashes by counting anymore?

Reply:
OK, I think I (finally) found out why this behavior was confusing me: As a fallback for the files without pageID or where `pageId not in ocrd_file.ID`, I have used the IDs in `ocrd_file.fileGrp`, not the `output_file_grp`.

I've dropped the clash-evading clause you mention, replaced it with a FileExistsError and adapted the test in ac1cc0805, now it does make sense to me.

Reply:
No more "`make_file_id` trying to be smart", the file IDs it generates will now always be either `{file_grp}_{file_id}` or `{file_grp}_{page_id}` (with mitigation for invalid `xs:ID`). It's up to the caller to ensure that the ID is actually unique, e.g. by appending a string. If it is not unique that means either bad input data (which we shouldn't try to work around but rather fail explicitly) or a bug in the processor (which should be fixed).